lib/Resource.js

```diff
@@ -32,7 +32,11 @@ Resource.prototype._construct = function(rawResource, client) {
 };
 
 Resource.prototype._checkIsResource = function(resource) {
-  if (resource instanceof Resource) return;
+  if (resource instanceof Resource) {
+    if (!resource._base.id) throw new Error("Resource has not been created remotely, it can't be assigned to a relationship!");
+    return;
```

Comment:
The rest of this function is error handling for the condition when the object passed is NOT an instance of Resource - we try and generate a helpful error message by interrogating the type of the object.